### PR TITLE
feat(Card): add link text option to clickable card

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

[Card] Add link text #2196

### Description

Entire card should be clickable. The PR removes the "static" type option that existed previously which made only the link within the card clickable - so the `type` prop is no longer necessary. Logic that differentiates the "static" and "link" type has been removed as well.

Option to add text to the Card cta has been added.

### Changelog

**New**

- add option for icon placement as video card renders the play icon on the left of the cta text (cta.iconPlacement)

**Changed**

- remove "link" type being passed into Card component from parent components

**Removed**

- `type` prop from the Card component

### CODE REVIEW

File: packages/react/src/components/Card/__stories__/Card.stories.js [hunk 1]
```diff
@@ -35,11 +35,11 @@ export default {
         cta: object(
```
Comment (jeffchew) [inline packages/react/src/components/Card/__stories__/Card.stories.js:35]: @annawen1 since under the hood, can this be broken out as separate knobs?

**Comment by jeffchew** [comment]
See this happening in percy, was this intentional?

Screen Shot 2020-07-09 at 2 56 40 PM

**Comment by annawen1** [comment]
> See this happening in percy, was this intentional?
> 
> Screen Shot 2020-07-09 at 2 56 40 PM

oh no, let me get that fixed

File: packages/react/src/components/Card/__stories__/Card.stories.js (hunk 1 above)
Comment (annawen1) [reply packages/react/src/components/Card/__stories__/Card.stories.js]: yep done

**Comment by jeffchew** [comment]
@annawen1 unfortunately more components got hit with the latest change:
https://percy.io/IBM-Digital-Design/IBM.com-Library-React/builds/6082707?utm_campaign=IBM-Digital-Design&utm_content=IBM.com-Library-React&utm_source=github_status_public

**Comment by jeffchew** [comment]
So close! It looks like LocaleModal was still affected:

Screen Shot 2020-07-10 at 5 39 20 PM

File: packages/styles/scss/components/card/index.scss [hunk 2]
```diff
@@ -23,7 +23,7 @@
       height: 100%;
       display: flex;
       flex-direction: column;
-      padding: $carbon--spacing-05 $carbon--spacing-05 0;
+      padding: $carbon--spacing-05;
```
Comment (diegomarcuz) [inline packages/styles/scss/components/card/index.scss]: I think what Jeff just mentioned It is happening because you are setting padding to all sides

Comment (annawen1) [reply packages/styles/scss/components/card/index.scss]: That code `padding: $carbon--spacing-05;` is already in the codebase, I edited it earlier to match the some specs but I reverted it and am going to speak to designer first

**Pull request merged**